src/world/Action/Action.cpp

```diff
@@ -809,11 +788,40 @@ Sapphire::Entity::CharaPtr Action::Action::getHitChara()
 
 bool Action::Action::hasValidLutEntry() const
 {
-  return m_lutEntry.potency != 0 || m_lutEntry.comboPotency != 0 || m_lutEntry.flankPotency != 0 || m_lutEntry.frontPotency != 0 ||
-    m_lutEntry.rearPotency != 0 || m_lutEntry.curePotency != 0 || m_lutEntry.restoreMPPercentage != 0;
+  return m_lutEntry.damagePotency != 0 || m_lutEntry.healPotency != 0 || m_lutEntry.selfHealPotency != 0 || m_lutEntry.selfStatus != 0 ||
+    m_lutEntry.targetStatus != 0 || m_lutEntry.gainMPPercentage != 0;
 }
 
 Action::EffectBuilderPtr Action::Action::getEffectbuilder()
 {
   return m_effectBuilder;
+}
+
+Data::ActionPtr Action::Action::getActionData() const
+{
+  return m_actionData;
+}
+
+bool Action::Action::isPhysical() const
+{
+  return isAttackTypePhysical( m_actionData->attackType );
+}
+
+bool Action::Action::isMagical() const
+{
+  return isAttackTypeMagical( m_actionData->attackType );
+}
+
+bool Action::Action::isAttackTypePhysical( int8_t attackType )
+{
+  return attackType == -1 ||
+         attackType == 1 ||
+         attackType == 2 ||
+         attackType == 3 ||
+         attackType == 4;
```

Comment:
Might wanna enum that

Reply:
exd structs should be generated however didn't find this one not sure if we should manually create enum or fix the generator

Reply:
Ah, said struct only needs to be added to the enum generator then if it is present as exd.

Reply:
https://github.com/SapphireServer/Sapphire/blob/master/src/tools/exd_common_gen/main.cpp#L117

Reply:
Fixed but the exd data maybe a little outdated and contains values that do not exist anymore. And it is in Japanese only so I had to manually edit it anyway.

Reply:
I wouldn't put it inside commongen if you need to change it each time its regenerated, that defeats the purpose of it. just put it in common and remove it from the generator otherwise it'll just break